spec/bigcommerce/prometheus/collectors/base_spec.rb

```diff
@@ -22,6 +22,18 @@
   let(:client) { double(:client, send_json: true) }
   let(:collector) { AppCollector.new(client: client, frequency: 0) }
 
+  describe '#start' do
+    subject { AppCollector.start(client: client, frequency: 0) }
+
+    it 'starts the collector on a thread' do
+      thread = subject
```

Comment:
Why the assignment? Doesn't `subject` memoize already?

Reply:
yeah, but I wanted to ensure any changes upstream in rspec (such as memoization alteration) wouldn't affect the process in question here, given it is around thread execution. Was more defensive coding than anything.